### PR TITLE
chore: remove psp

### PR DESCRIPTION
**Pull request by aramase**
Signed-off-by: Anish Ramasekar <anish.ramasekar@gmail.com>

**Reason for Change**:

- Removes deprecated psp

**Requirements**

- [x] squashed commits
- [x] included documentation
- [x] added unit tests and e2e tests (if applicable).

**Issue Fixed**:

fixes https://github.com/Azure/secrets-store-csi-driver-provider-azure/issues/787

**Does this change contain code from or inspired by another project?**

- [ ] Yes
- [ ] No

**If "Yes," did you notify that project's maintainers and provide attribution?**

**Special Notes for Reviewers**:

### CODE REVIEW

File: manifest_staging/charts/csi-secrets-store-provider-azure/templates/podsecuritypolicy.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{{- if .Values.rbac.pspEnabled }}
+{{- if and .Values.rbac.pspEnabled (.Capabilities.APIVersions.Has "policy/v1beta1/PodSecurityPolicy") }}
```
Comment (nilekhc) [inline manifest_staging/charts/csi-secrets-store-provider-azure/templates/podsecuritypolicy.yaml:1]: Do we want to support this till 1.24 EOL?

Comment (aramase) [reply manifest_staging/charts/csi-secrets-store-provider-azure/templates/podsecuritypolicy.yaml:1]: We're supporting it with this change. We're checking if the cluster still has the capability, and if yes, we're deploying the pod security policy.

Comment (nilekhc) [reply manifest_staging/charts/csi-secrets-store-provider-azure/templates/podsecuritypolicy.yaml:1]: I asked the wrong question... 😅. But I got the answer. After 1.24 EOL we can completely remove this. Thanks.

**Review by nilekhc — APPROVED** [review]

**Comment by aramase** [comment]
> @aramase do you know if there are any plans to release with these changes any time soon? These changes aren't in the latest version (1.3.0) and as a result we're not currently able to install on the latest Kubernetes version (1.25) which removes support for PodSecurityPolicys. Thanks!

@graemerobertson The charts will be released this week. Pod Security Policy is only installed when `rbac.pspEnabled=true`. You should be able to install the helm charts without that flag which is default false.

**Comment by graemerobertson** [comment]
Out of interest, are there any plans to resolve https://github.com/Azure/secrets-store-csi-driver-provider-azure/issues/967? 🙂

**Pull request merged**